sdk/python/kubeflow/storage_initializer/s3.py

```diff
@@ -42,18 +42,31 @@ def download_dataset(self):
         import boto3
 
         # Create an S3 client for Nutanix Object Store/S3
-        s3_client = boto3.client(
-            "s3",
+        s3_client = boto3.Session(
             aws_access_key_id=self.config.access_key,
             aws_secret_access_key=self.config.secret_key,
-            endpoint_url=self.config.endpoint_url,
```

Comment:
What is the reason that we should pass the `endpoint_url` into the `s3_client.resource`?

Reply:
@tenzen-y  With endpoint urls, it can work with any S3 protocol compliant implementations

Reply:
That makes sense.
Thanks!